src/main/kotlin/com/katanox/tabour/extentions/Retry.kt

```diff
@@ -0,0 +1,26 @@
+package com.katanox.tabour.extentions
+
+import kotlinx.coroutines.delay
+import mu.KotlinLogging
+
+val logger = KotlinLogging.logger {}
+
+suspend inline fun <T> retry(
+    times: Int = 1,
+    initialDelay: Long = 100, // 0.1 second
+    maxDelay: Long = 1000, // 1 second
+    factor: Double = 2.0,
+    block: () -> T,
+): T {
+    var currentDelay = initialDelay
+    repeat(times - 1) {
```

Comment:
i think that currently this function will execute `N times` the function, even if it succeeds the first time.

I guess we need a flag that indicates that the function has already been executed succefully, change repeat to `repeat(times)` and use the flag to execute:
```
repeat(times) {
  if (!hasExecutedSuccesfully) ...
}
```

Reply:
yeah that is valid